gdk4/src/event.rs

```diff
@@ -0,0 +1,616 @@
+        unsafe { from_glib_none(ffi::gdk_event_get_device(self.to_glib_none().0)) }
+    }
+
+    #[doc(alias = "gdk_event_get_display")]
+    pub fn get_display(&self) -> Option<Display> {
+        unsafe { from_glib_none(ffi::gdk_event_get_display(self.to_glib_none().0)) }
+    }
+
+    #[doc(alias = "gdk_event_get_event_type")]
+    pub fn get_event_type(&self) -> EventType {
+        unsafe { from_glib(ffi::gdk_event_get_event_type(self.to_glib_none().0)) }
+    }
+
+    #[doc(alias = "gdk_event_get_history")]
+    pub fn get_history(&self) -> Vec<TimeCoord> {
+        unsafe {
+            let mut out_n_coords = mem::MaybeUninit::uninit();
+            let ret = FromGlibContainer::from_glib_container_num(
+                ffi::gdk_event_get_history(self.to_glib_none().0, out_n_coords.as_mut_ptr()),
+                out_n_coords.assume_init() as usize,
+            );
+            ret
+        }
+    }
+
+    #[doc(alias = "gdk_event_get_modifier_state")]
+    pub fn get_modifier_state(&self) -> ModifierType {
+        unsafe { from_glib(ffi::gdk_event_get_modifier_state(self.to_glib_none().0)) }
+    }
+
+    #[doc(alias = "gdk_event_get_pointer_emulated")]
+    pub fn get_pointer_emulated(&self) -> bool {
+        unsafe { from_glib(ffi::gdk_event_get_pointer_emulated(self.to_glib_none().0)) }
+    }
+
+    #[doc(alias = "gdk_event_get_position")]
+    pub fn get_position(&self) -> Option<(f64, f64)> {
+        unsafe {
+            let mut x = mem::MaybeUninit::uninit();
+            let mut y = mem::MaybeUninit::uninit();
+            let ret = from_glib(ffi::gdk_event_get_position(
+                self.to_glib_none().0,
+                x.as_mut_ptr(),
+                y.as_mut_ptr(),
+            ));
+            if ret {
+                let x = x.assume_init();
+                let y = y.assume_init();
+                Some((x, y))
+            } else {
+                None
+            }
+        }
+    }
+
+    #[doc(alias = "gdk_event_get_seat")]
+    pub fn get_seat(&self) -> Option<Seat> {
+        unsafe { from_glib_none(ffi::gdk_event_get_seat(self.to_glib_none().0)) }
+    }
+
+    #[doc(alias = "gdk_event_get_surface")]
+    pub fn get_surface(&self) -> Option<Surface> {
+        unsafe { from_glib_none(ffi::gdk_event_get_surface(self.to_glib_none().0)) }
+    }
+
+    #[doc(alias = "gdk_event_get_time")]
+    pub fn get_time(&self) -> u32 {
+        unsafe { ffi::gdk_event_get_time(self.to_glib_none().0) }
+    }
+
+    #[doc(alias = "gdk_event_triggers_context_menu")]
+    pub fn triggers_context_menu(&self) -> bool {
+        unsafe { from_glib(ffi::gdk_event_triggers_context_menu(self.to_glib_none().0)) }
+    }
+}
+
+impl fmt::Display for Event {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        f.write_str("Event")
+    }
+}
+
+pub trait EventTypeTrait: StaticType + FromGlibPtrFull<*mut ffi::GdkEvent> + 'static {
+    fn event_types() -> Vec<EventType>;
```

Comment:
```suggestion
    fn event_types() -> &'static [EventType];
```

Or not? This won't change at runtime :)